src/library-authoring/components/PublishConfirmationModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { useEffect } from 'react';
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
+import { Alert, Button, useToggle } from '@openedx/paragon';
+
+import BaseModal from '../../editors/sharedComponents/BaseModal';
+import messages from './messages';
+import infoMessages from '../component-info/messages';
+import { ComponentUsage } from '../component-info/ComponentUsage';
+
+interface PublishConfirmationModalProps {
+  isOpen: boolean,
+  onClose: () => void,
+  onConfirm: () => void,
+  displayName: string,
+  usageKey: string,
+  showDownstreams: boolean,
+}
+
+interface PublishConfirmationBodyProps {
+  displayName: string,
+}
+
+const PublishConfirmationBody = ({
+  displayName,
+}: PublishConfirmationBodyProps) => {
+  const intl = useIntl();
+  return (
+    <div className="pt-4">
+      {intl.formatMessage(messages.publishConfirmationBody)}
+      <div className="mt-2 p-2 border">
+        {displayName}
+      </div>
+    </div>
+  );
+};
+
+const PublishConfirmationModal = ({
+  isOpen,
+  onClose,
+  onConfirm,
+  usageKey,
+  displayName,
+  showDownstreams,
+}: PublishConfirmationModalProps) => {
+  const intl = useIntl();
+
+  const [
+    isDownstreamsEmpty,
+    setDownstreamsEmpty,
+    setDownstreamsNotEmpty,
+  ] = useToggle(false);
+
+  useEffect(() => {
+    setDownstreamsNotEmpty();
+  }, [usageKey]);
+
```

Comment:
Do we need this?

Reply:
Yes. When you try to publish a component already published but without downstream, the modal shows the empty state, and later when you try to publish a component already published with downstream, the modal also shows the empty state. With this, we return the value to the default. I added a comment: https://github.com/openedx/frontend-app-authoring/pull/1677/commits/bf440d6ad93a290eac711a5223db70b61935b896